### ISSUE TITLE
Uncaught TypeError: Cannot read property '0' of null

### ISSUE THREAD

**Issue by ghost** [ajdm/linter-matlab#4]
Installed linter and linter-matlab packages.
mlint.exe is on system path.
Trying to lint did nothing (no errors either).

Added mlint.exe directory in linter-matlab settings.
Linting still doesn't work but following error is thrown upon saving a file:

**Atom Version**: 1.2.0
**System**: Microsoft Windows 10 Pro
**Thrown From**: [linter-matlab](https://github.com/ajdm/linter-matlab) package, v1.0.0
### Stack Trace

Uncaught TypeError: Cannot read property '0' of null

```
At /C:/Users/Ben/.atom/packages/linter-matlab/lib/linter-matlab.coffee:49

TypeError: Cannot read property '0' of null
    at BufferedProcess.stderr (file:///C:/Users/Ben/.atom/packages/linter-matlab/lib/linter-matlab.coffee:40:10)
    at Socket.<anonymous> (C:\Users\Ben\AppData\Local\atom\app-1.2.0\resources\app.asar\src\buffered-process.js:85:13)
    at emitOne (events.js:77:13)
    at Socket.emit (events.js:169:7)
    at readableAddChunk (_stream_readable.js:146:16)
    at Socket.Readable.push (_stream_readable.js:110:10)
    at Pipe.onread (net.js:523:20)
```
### Commands

```
  2x -2:00.9.0 editor:newline (atom-text-editor.editor.is-focused)
     -1:56.4.0 core:backspace (atom-text-editor.editor.is-focused)
     -1:54.2.0 core:delete (atom-text-editor.editor.is-focused)
     -1:53.9.0 core:move-down (atom-text-editor.editor.is-focused)
     -1:52.2.0 window:toggle-dev-tools (atom-text-editor.editor.is-focused)
  5x -1:17.1.0 core:backspace (atom-text-editor.editor.is-focused)
     -1:14.9.0 editor:newline (atom-text-editor.editor.is-focused)
  4x -1:13.9.0 core:delete (atom-text-editor.editor.is-focused)
     -1:12.3.0 core:move-up (atom-text-editor.editor.is-focused)
     -1:12 editor:move-to-end-of-screen-line (atom-text-editor.editor.is-focused)
     -1:10.3.0 core:move-down (atom-text-editor.editor.is-focused)
 22x -1:01.7.0 core:undo (a.inline-block)
  2x -0:45 linter:next-error (atom-workspace.workspace.scrollbars-visible-always.theme-base16-syntax.theme-atom-material-ui)
     -0:30.5.0 core:move-down (atom-text-editor.editor.is-focused)
     -0:04.2.0 core:delete (atom-text-editor.editor.is-focused)
     -0:01.1.0 core:save (atom-text-editor.editor.is-focused)
```
### Config

``` json
{
  "core": {
    "autoHideMenuBar": true,
    "projectHome": "C:\\DICE\\source",
    "themes": [
      "atom-material-ui",
      "base16-syntax"
    ]
  },
  "linter-matlab": {
    "mlintDir": "C:\\Program Files\\MATLAB\\R2015a\\bin\\win64"
  }
}
```
### Installed Packages

``` coffee
# User
atom-material-ui, v0.8.0
auto-detect-indentation, v0.4.2
base16-syntax, v1.5.0
highlight-selected, v0.10.1
language-matlab, v0.1.1
language-mediawiki, v2.0.0
less-than-slash, v0.11.0
linter, v1.11.1
linter-matlab, v1.0.0
minimap, v4.16.0
minimap-highlight-selected, v4.3.1
pigments, v0.18.1
sync-settings, v0.6.0

# Dev
No dev packages
```

**Comment by jdmanton** [comment]
Thanks for the bug report. I'm afraid I don't know much about Atom package development, so it may take me a while to fix this, especially as I don't have ready access to a Windows machine at the moment (although I do have the plugin working correctly on a Windows machine).

If you change line 27 in `linter-matlab.coffee` to

``` coffeescript
editor = atom.workspace.getActivePaneItem()
filePath = editor?.buffer.file
```

does this fix the problem?

**Comment by ghost** [comment]
Thanks for the very quick reply.
After making the change you suggested it no longer throws an error when saving the file, but the linter doesn't work either. It lists correctly as "linter-matlab loaded" in the console but other than that it doesn't appear to be doing anything, or at least it doesn't generate any output.

Edit:
I tested a Python linter on my system (linter-flake8) and that works fine. So the problem appears to be specific to linter-matlab.

**Comment by jdmanton** [comment]
Strangely, if I make the changes I suggested on my work Windows computer, I am given an error message every time I save the file and the linter doesn't work.

**Comment by jdmanton** [comment]
Ah, I think I've solved it! Your `mlintDir` value doesn't have a trailing slash, which it needs. I'll add a check for this that adds one if it's missing and push an update. Thanks.

**Comment by jdmanton** [comment]
If you upgrade to v1.1.0 of `linter-matlab`, this issue should now be fixed.

**Comment by ghost** [comment]
Perfect, that did it. Thank you very much!

**Comment by jdmanton** [comment]
Great! I hope you find the linter useful.

**Comment by r0oland** [comment]
Thanks for solving this issue! Had the same problem. 

I recommend you add the following text to your install instructions:
"The directory containing mlint.exe (NOT mlint.m) which on Windows machines is typically "C:\Program Files\MATLAB[your_matlab_version]\bin\win64\"
E.g. "C:\Program Files\MATLAB\R2016a\bin\win64\"

That will help lots of people actually getting this to work. Now it works like a charm and makes my life with Matlab and Atom so much easier!

**Comment by jdmanton** [comment]
Thanks, I've merged in your pull request (#7) that does this.
